app/controllers/comment_controller.rb

```diff
@@ -191,7 +191,9 @@ def prev_comment
   def add_comment
     pass_query_params
     @target = Comment.find_object(params[:type], params[:id].to_s)
-    if !allowed_to_see!(@target)
+    if !@target
+      redirect_back_or_default("/")
+    elsif !allowed_to_see!(@target)
```

Comment:
Would this work:
```suggestion
    redirect_back_or_default("/") unless @target
    
    if !allowed_to_see!(@target)
```

Reply:
Not quite, we'd need to return, too.  But look at the whole method, it's badly in need of renormalizing.  I figured it was better to stick with the style of the rest of the method for such a tiny, inconsequential bug fix.  Sorry, I should have explicitly stated that this PR is not worth wasting time on!